tests/bin/sessions.test.ts

```diff
@@ -1,215 +1,322 @@
 import Logger, { LogLevel, StreamHandler } from '@matrixai/logger';
-import PolykeyAgent from '@/PolykeyAgent';
+import { Session } from '@/sessions';
 import { sleep } from '@/utils';
-import * as testUtils from './utils';
-
-jest.mock('@/keys/utils', () => ({
-  ...jest.requireActual('@/keys/utils'),
-  generateDeterministicKeyPair:
-    jest.requireActual('@/keys/utils').generateKeyPair,
-}));
+import config from '@/config';
+import * as clientErrors from '@/client/errors';
+import * as testBinUtils from './utils';
 
 /**
- * This test file has been optimised to use only one instance of PolykeyAgent where posible.
- * Setting up the PolykeyAgent has been done in a beforeAll block.
- * Keep this in mind when adding or editing tests.
- * Any side effects need to be undone when the test has completed.
- * Preferably within a `afterEach()` since any cleanup will be skipped inside a failing test.
- *
- * - left over state can cause a test to fail in certain cases.
- * - left over state can cause similar tests to succeed when they should fail.
- * - starting or stopping the agent within tests should be done on a new instance of the polykey agent.
- * - when in doubt test each modified or added test on it's own as well as the whole file.
- * - Looking into adding a way to safely clear each domain's DB information with out breaking modules.
+ * Mock prompts module which is used prompt for password
  */
+jest.mock('prompts');
+const mockedPrompts = mocked(prompts);
 
-describe('Session Token Refreshing', () => {
-  const logger = new Logger('pkStdio Test', LogLevel.WARN, [
+describe('CLI Sessions', () => {
+  const logger = new Logger('status test', LogLevel.WARN, [
     new StreamHandler(),
   ]);
   let dataDir: string;
-  let nodePath: string;
-  let passwordFile: string;
-  let sessionFile: string;
-  let polykeyAgent: PolykeyAgent;
-
-  let tokenBuffer: Buffer;
-  let token: SessionToken;
-  let command: string[];
-  const vaultName = 'TestVault' as VaultName;
-
+  let pkAgentClose;
+  let sessionTokenPath: string;
   beforeAll(async () => {
-    // This handles the expensive setting up of the polykey agent.
+    sessionTokenPath = path.join(global.binAgentDir, config.defaults.tokenBase);
+    pkAgentClose = await testBinUtils.pkAgent();
+  }, global.maxTimeout);
+  afterAll(async () => {
+    await pkAgentClose();
+  });
+  beforeEach(async () => {
     dataDir = await fs.promises.mkdtemp(
       path.join(os.tmpdir(), 'polykey-test-'),
     );
-    nodePath = path.join(dataDir, 'keynode');
-    command = ['vaults', 'list', '-np', nodePath];
-    passwordFile = path.join(dataDir, 'passwordFile');
-    sessionFile = path.join(nodePath, 'token');
-    await fs.promises.writeFile(passwordFile, 'password');
-    polykeyAgent = await PolykeyAgent.createPolykeyAgent({
-      password: 'password',
-      nodePath: nodePath,
-      logger: logger,
+    // Invalidate all active sessions
+    await testBinUtils.pkStdio(
+      ['agent', 'lock'],
```

Comment:
Forgot that this uses `agent lock`. If so then we can put these tests into the `tests/bin/agent` directory.

Also you shouldn't only do this test after first establishing a session. Unless the assumption is that you need to ensure that the session token is destroyed before running any test. In that case you should put it into `beforeEach`.

Reply:
Oh I just realised this _is_ in `beforeEach`nvm.

Reply:
Will move it over to `tests/bin/agent`. Test file still called `sessions.test.ts` I'm assuming?

Reply:
Yea.